Subject: [PATCH] ppc64: offb remapped address
Patch-mainline: 2.6.12-rc1

From: Jake Moilanen <moilanen@austin.ibm.com>

The offb code did not take into account a remapped pci address.  Adding in
the pci_mem_offset fixed a DSI in offb.

Signed-off-by: Jake Moilanen <moilanen@austin.ibm.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Olaf Hering <olh@suse.de>
---

 25-akpm/drivers/video/offb.c |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletion(-)

diff -puN drivers/video/offb.c~ppc64-offb-remapped-address drivers/video/offb.c
--- 25/drivers/video/offb.c~ppc64-offb-remapped-address	Thu Mar  3 16:50:19 2005
+++ 25-akpm/drivers/video/offb.c	Thu Mar  3 16:50:19 2005
@@ -29,6 +29,10 @@
 #include <asm/io.h>
 #include <asm/prom.h>
 
+#ifdef CONFIG_PPC64
+#include <asm/pci-bridge.h>
+#endif
+
 #ifdef CONFIG_PPC32
 #include <asm/bootx.h>
 #endif
@@ -322,7 +326,8 @@ static void __init offb_init_nodriver(st
 	int *pp, i;
 	unsigned int len;
 	int width = 640, height = 480, depth = 8, pitch;
-	unsigned *up, address;
+	unsigned *up;
+	unsigned long address;
 
 	if ((pp = (int *) get_property(dp, "depth", &len)) != NULL
 	    && len == sizeof(int))
@@ -357,6 +362,10 @@ static void __init offb_init_nodriver(st
 
 		address = (u_long) dp->addrs[i].address;
 
+#ifdef CONFIG_PPC64
+		address += dp->phb->pci_mem_offset;
+#endif
+
 		/* kludge for valkyrie */
 		if (strcmp(dp->name, "valkyrie") == 0)
 			address += 0x1000;
_
